### DISCUSSION TITLE
Support for SSO Provider with Authorization Code Flow with PKCE

### DISCUSSION THREAD

**Discussion by chriskhanhtran-verisk** [langfuse/langfuse#4800] [Support] [answered]
I am integrating my Okta application, which uses the authorization code flow with PKCE, into your platform. To enable seamless SSO integration, I need two options in the OktaProvider configuration:

[code](https://github.com/langfuse/langfuse/blob/8fd93ecff9b87fd7cff78f8124b6077033deb3c4/web/src/server/auth.ts#L211)
```
OktaProvider({
    checks: ['pkce'],
    client: { token_endpoint_auth_method: 'none' },
    ...
}),
```

Currently, I noticed the customSsoProvider includes `checks: ["pkce", "state"]` ([code](https://github.com/langfuse/langfuse/blob/8fd93ecff9b87fd7cff78f8124b6077033deb3c4/packages/shared/src/server/auth/customSsoProvider.ts#L19)), which works for configuring checks. However, there’s no way to specify `client: { token_endpoint_auth_method: 'none' }` in the provider options.

References:
- [Okta OAuth Guide](https://developer.okta.com/docs/api/openapi/okta-oauth/guides/client-auth/)
- [NextAuth.js Issue #3466](https://github.com/nextauthjs/next-auth/issues/3466)
- [Okta PKCE Integration Forum Post](https://devforum.okta.com/t/next-auth-with-okta-pkce-flow/26312)

Could you please add support for configuring checks and token_endpoint_auth_method via environment variables? PKCE is a widely-used flow for OAuth 2.0 and OpenID Connect, and adding this functionality would improve compatibility with Okta and other similar providers.

Thank you for considering this enhancement!

### Additional information

Okta application config:
![image](https://global.discourse-cdn.com/oktadev/original/2X/3/384a8a20e3899c7b2dcdec553a5c59df4af1738e.png)

### SDK and container versions
Image: langfuse/langfuse:3

**Comment by marcklingen** [comment, marked as answer]
thanks for creating the issue to track this!
https://github.com/langfuse/langfuse/issues/4799
